tests/python/pants_test/subsystem/test_subsystem.py

```diff
@@ -266,3 +275,17 @@ def subsystem_dependencies(cls):
 
     with self.assertRaises(SubsystemClientMixin.CycleException):
       list(SubsystemB.subsystem_closure_iter())
+
+  def test_get_streaming_workunit_callbacks(self):
+    import_str = "pants_test.subsystem.test_subsystem.WorkunitSubscriptableSubsystem"
+    callables_list = Subsystem.get_streaming_workunit_callbacks([import_str])
+    assert len(callables_list) == 1
+
+  def test_streaming_workunit_callbacks_with_invalid_subsystem(self):
+    import_str = "pants_test.subsystem.test_subsystem.DummySubsystem"
+    with self.captured_logging(level = logging.WARNING) as captured:
+      callables_list = Subsystem.get_streaming_workunit_callbacks([import_str])
+      warnings = captured.warnings()
+      assert len(warnings) == 1
+      assert "does not have a method named `handle_workunits` defined" in warnings[0]
```

Comment:
```suggestion
      self.assertIn("does not have a method named `handle_workunits` defined", 
                    assert_single_element(captured.warnings()))
```

Reply:
Ah, so we only started using this `assert` idiom recently (although Pex has always used it). Normally, this would be a bad idea. But, Pytest has magic over assert statements that make the output much nicer upon failure than unittest style assertions.

See https://docs.pytest.org/en/latest/assert.html. Specifically,

```
______________________________ test_function _______________________________

    def test_function():
>       assert f() == 4
E       assert 3 == 4
E        +  where 3 = f()

test_assert1.py:6: AssertionError
```

Pytest adds statements like ` where 3 = f()` to show how each value was derived, but can only do that with `assert`.